docs/udf.rst

```diff
@@ -0,0 +1,179 @@
+you should create an "uber-jar". The classes in your uber-jar will be loaded in preference to any
+on the KSQL classpath excluding anything that is vital to the running of KSQL, i.e., classes that are
+part of ``org.apache.kafka`` and ``io.confluent``. Further, the ``ClassLoader`` can restrict access
+to other classes via a blacklist. The blacklist file is ``resource-blacklist.txt``. You can add
+any classes or packages that you wan't blacklisted from UDF use, for example you probably don't
+want a UDF to be able to fork processes or shutdown the JVM.
+
+=============
+Creating UDFs
+=============
+
+To create a UDF you need to create a class that is annotated with ``@UdfDescription``. Each method in
+the class that you want to represent a UDF must be public and annotated with ``@Udf``. The class
+you create represents a collection of UDFs all with the same name but may have different
+arguments and return types.
+
+
+UdfDescription Annotation
+-------------------------
+The ``@UdfDescription`` annotation is applied at the class level and has four fields, 2 of which are required:
+
++------------+------------------------------+---------+
+| Field      | Description                  | Required|
++============+==============================+=========+
+| name       | The name of the function(s)  | Y       |
+|            | represented by this class.   |         |
++------------+------------------------------+---------+
+| description| A string describing generally| Y       |
+|            | what the function(s) in this |         |
+|            | class do.                    |         |
++------------+------------------------------+---------+
+| author     | The author of the UDF.       | N       |
++------------+------------------------------+---------+
+| version    | The version of the UDF.      | N       |
++------------+------------------------------+---------+
+
+
+Udf Annotation
+--------------
+
+The ``@Udf`` annotation is applied to public methods of a class annotated with ``@UdfDescription``.
+Each annotated method will become an invocable function in KSQL. The annotation only has a single
+field ``description`` that is required. You can use this to better describe what a particular version
+of the UDF does, for example:
+
+.. code:: java
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the end of this string.")
+    public String substring(final String value, final int startIndex)
+
+    @Udf(description = "Returns a string that is a substring of this string. The"
+        + " substring begins with the character at the specified startIndex and"
+        + " extends to the character at endIndex -1.")
+    public String substring(final String value, final int startIndex, final int endIndex)
+
+
+Example UDF class
```

Comment:
I would move the example to the top and then break it down through the individual sections. Would be easier to read because the example gives the reader an anchor point for all the details.

Reply:
Thanks!